### ISSUE TITLE
`use:enhance` doesnt mirror native form behaviour for default values

### ISSUE THREAD

**Issue by sacrosanctic** [sveltejs/kit#11971]
### Describe the bug

```svelte
<form method="post" use:enhance>
  <input type="text" value="text" />
  <button>submit</button>
</form>
```

## Expected
`on:submit`, input resets back to `"text"`

## Actual
`on:submit`, input resets to `""`

### with two way binding
```svelte
<script>
  let value = "text"
</script>

<form method="post" use:enhance={({ formElement })=>{
  return (async( {update} )=>{
    await update()
    value = 'text'
  })
}}>
  <input type="text" bind:value/>
  <button>submit</button>
</form>
```
This works but only if the input value has changed, otherwise, it resets to `""`.

---
An additional workaround produces the desired result
```diff
<script>
  let value = "text"
</script>

<form method="post" use:enhance={({ formElement })=>{
  return (async( {update} )=>{
    await update()
+    value = ''
    value = 'text'
  })
}}>
  <input type="text" bind:value/>
  <button>submit</button>
</form>
```

### Reproduction

https://www.sveltelab.dev/p6trf8xmtubbn07

### Severity

annoyance

**Comment by dummdidumm** [comment]
The problem is that `value` serves as both the current value (for SSR) and the default value. Related https://github.com/sveltejs/svelte/pull/10617

**Comment by brunnerh** [comment]
I actually would have expected the reset keep the value if SSR is not disabled since the attribute should set the default value. So this feels like a problem caused by hydration because it does not work even on the first SSR-based load.

Related:
- https://github.com/sveltejs/svelte/issues/8220

**Comment by brunnerh** [comment]
Oh, there appears to be a [dev mode issue](https://github.com/sveltejs/svelte/issues/8220#issuecomment-1400559713) here, in preview it works for me...

**Comment by sacrosanctic** [comment]
> The problem is that `value` serves as both the current value (for SSR) and the default value. Related [sveltejs/svelte#10617](https://github.com/sveltejs/svelte/pull/10617)

I think there could be two issues here, my initial examples do not use two-way binding. Yet, the default value is not preserved.

**Comment by ITenthusiasm** [comment]
https://github.com/sveltejs/svelte/issues/9230 might be related as well. Resolving that issue would automatically resolve https://github.com/sveltejs/svelte/issues/8220. If there was a way to be distinct about the `defaultValue` (or, more accurately, to be distinct about the _attributes_ vs. the _properties_), problems like this one would probably be solved automatically as well, I believe.

[Solid.js](https://www.solidjs.com/docs/latest/api#prop___) allows developers to distinguish between properties on elements and attributes on elements. And it accomplishes this using directive-like syntax. If Svelte had something like this, it would be a _huge_ win.

**Comment by theetrain** [comment]
I've faced a closely related issue on production, but I couldn't come up with a reproduction. This is the closest I got: https://www.sveltelab.dev/65dr62xmk7ct6iy

The unexpected behaviour was:

1. User goes to page
2. `load` returns initial values for form inputs, which initializes our inputs via `value={data.something}`
3. User changes input contents, then clicks 'submit'
4. :x: Inputs with changed values immediately switch back to their initial value
5. Form action completes, `update()` gets called, and user redirects to another page (per form action logic)

After much struggling, I believe it was `enhance` that was causing our forms to reset to their initial value.

However, we upgraded from `svelte@4.2.19` to `svelte@5.1.0` and the issue went away. I had assumed `enhance` was the issue, but perhaps the way it calls `resetForm` and the way Svelte handles that has changed.
